src/sentry/api/event_search.py

```diff
@@ -94,45 +94,55 @@
 aggregate_rel_date_filter = negation? aggregate_key sep operator? rel_date_format
 
 # has filter for not null type checks
-has_filter = negation? "has" sep (search_key / search_value)
+has_filter = negation? &"has:" search_key sep (search_key / search_value)
 
 # is filter. Specific to issue search
-is_filter  = negation? "is" sep search_value
+is_filter  = negation? &"is:" search_key sep search_value
 
 # in filter key:[val1, val2]
 text_in_filter = negation? text_key sep text_in_list
 
 # standard key:val filter
 text_filter = negation? text_key sep search_value
 
-key              = ~r"[a-zA-Z0-9_\.-]+"
-quoted_key       = ~r"\"([a-zA-Z0-9_\.:-]+)\""
+key              = ~r"[a-zA-Z0-9_.-]+"
+quoted_key       = '"' ~r"[a-zA-Z0-9_.:-]+" '"'
 explicit_tag_key = "tags" open_bracket search_key closed_bracket
 aggregate_key    = key open_paren spaces function_args? spaces closed_paren
 function_args    = key (spaces comma spaces key)*
 search_key       = key / quoted_key
 text_key         = explicit_tag_key / search_key
-value            = ~r"[^()\s]*"
-quoted_value     = ~r"\"((?:\\\"|[^\"])*)?\""s
-in_value         = ~r"[^(),\s]*(?:[^\],\s)]|](?=]))"
-text_value       = quoted_value / in_value
+value            = ~r"[^()\t\n ]*"
+quoted_value     = '"' ('\\"' / ~r'[^"]')* '"'
+in_value         = (&in_value_termination in_value_char)+
+text_in_value    = quoted_value / in_value
 search_value     = quoted_value / value
-numeric_value    = "-"? numeric ("k"/"m"/"b")? ~r"(?=\s|\)|$|,|])"
-boolean_value    = ~r"(true|1|false|0)(?=\s|\)|$)"i
-text_in_list     = open_bracket text_value (spaces comma spaces text_value)* closed_bracket
-numeric_in_list  = open_bracket numeric_value (spaces comma spaces numeric_value)* closed_bracket
+numeric_value    = "-"? numeric ~r"[kmb]"? &(end_value / comma / closed_bracket)
+boolean_value    = ~r"(true|1|false|0)"i &end_value
+text_in_list     = open_bracket text_in_value (spaces comma spaces text_in_value)* closed_bracket &end_value
+numeric_in_list  = open_bracket numeric_value (spaces comma spaces numeric_value)* closed_bracket &end_value
+
+# See: https://stackoverflow.com/a/39617181/790169
+in_value_termination = in_value_char (!in_value_end in_value_char)* in_value_end
+in_value_char        = ~r"[^(), ]"
+in_value_end         = closed_bracket / (spaces comma)
 
 # Formats
-iso_8601_date_format = ~r"(\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(\.\d{1,6})?)?(Z|([+-]\d{2}:\d{2}))?)(?=\s|\)|$)"
-rel_date_format      = ~r"[\+\-][0-9]+[wdhm](?=\s|\)|$)"
-duration_format      = numeric ("ms"/"s"/"min"/"m"/"hr"/"h"/"day"/"d"/"wk"/"w") ~r"(?=\s|\)|$)"
+date_format = ~r"\d{4}-\d{2}-\d{2}"
```

Comment:
Didn't carry over the `num2` `num4` stuff since like.. I think people can get it